node/channel.js

```diff
@@ -453,20 +454,37 @@ TChannel.prototype.makeSubChannel = function makeSubChannel(options) {
     return chan;
 };
 
-TChannel.prototype.listen = function listen(port, host, callback) {
+TChannel.prototype.listen = function listen(port, host, opts, callback) {
     // Note:
     // - 0 is a valid port number, indicating that the system must assign an
     //   available ephemeral port
     // - 127.0.0.1 is a valid host, primarily for testing
+    // - opts supports:
+    //   {fd: X} - listen will attach to the existing socket, and host/port
+    //   remain required, but become only descriptive. See also:
```

Comment:
The use case here passing the `fd` from the command line and sharing it across two processes for zero-downtime restarts.